testing/integration/fqdn_test.go

```diff
@@ -0,0 +1,292 @@
+			kibana.MonitoringEnabledMetrics,
+		},
+	}
+	policy, err := kibClient.CreatePolicy(createPolicyReq)
+	require.NoError(s.T(), err)
+
+	// Create enrollment API key
+	createEnrollmentApiKeyReq := kibana.CreateEnrollmentAPIKeyRequest{
+		PolicyID: policy.ID,
+	}
+	enrollmentToken, err := kibClient.CreateEnrollmentAPIKey(createEnrollmentApiKeyReq)
+	require.Nil(s.T(), err, "Could not create enrollment token")
+
+	// Get default Fleet Server URL
+	fleetServerURL, err := tools.GetDefaultFleetServerURL(kibClient)
+	require.NoError(s.T(), err)
+
+	// Enroll agent
+	output, err := tools.EnrollElasticAgent(fleetServerURL, enrollmentToken.APIKey, s.agentFixture)
+	if err != nil {
+		s.T().Log(string(output))
+	}
+	require.NoError(s.T(), err)
+
+	require.Eventually(s.T(), tools.WaitForAgentStatus(s.T(), kibClient, "online"), 2*time.Minute, 10*time.Second, "Agent status is not online")
+
+	// Verify that agent name is short hostname
+	agent, err := tools.GetAgentByHostnameFromList(kibClient, shortName)
+	require.NoError(s.T(), err)
+	require.NotNil(s.T(), agent)
+
+	// Verify that hostname in `logs-*` and `metrics-*` is short hostname
+	s.verifyHostNameInIndices("logs-*", shortName)
+	s.verifyHostNameInIndices("metrics-*", shortName)
+
+	// Update Agent policy to enable FQDN
+	policy.AgentFeatures = []map[string]interface{}{
+		{
+			"name":    "fqdn",
+			"enabled": true,
+		},
+	}
+	updatePolicyReq := kibana.UpdatePolicyRequest(*policy)
+	_, err = kibClient.UpdatePolicy(updatePolicyReq)
+	require.NoError(s.T(), err)
+
+	// Wait until policy has been applied by Agent
+	prevAgentPolicyRevision := agent.PolicyRevision
+	require.Eventually(s.T(), func() bool {
+		getAgentReq := kibana.GetAgentRequest{ID: agent.ID}
+		updatedPolicyAgent, err := kibClient.GetAgent(getAgentReq)
+		require.NoError(s.T(), err)
+
+		if updatedPolicyAgent.PolicyRevision > prevAgentPolicyRevision {
+			prevAgentPolicyRevision = updatedPolicyAgent.PolicyRevision
+			return true
+		}
+
+		return false
+	}, 30*time.Second, 1*time.Second)
+
+	// Verify that agent name is FQDN
+	agent, err = tools.GetAgentByHostnameFromList(kibClient, fqdn)
+	require.NoError(s.T(), err)
+	require.NotNil(s.T(), agent)
+
+	// Verify that hostname in `logs-*` and `metrics-*` is FQDN
+	s.verifyHostNameInIndices("logs-*", fqdn)
+	s.verifyHostNameInIndices("metrics-*", fqdn)
+
+	// Update Agent policy to disable FQDN
+	policy.AgentFeatures = []map[string]interface{}{
+		{
+			"name":    "fqdn",
+			"enabled": false,
+		},
+	}
+	updatePolicyReq = kibana.UpdatePolicyRequest(*policy)
+	_, err = kibClient.UpdatePolicy(updatePolicyReq)
+	require.NoError(s.T(), err)
+
+	// Wait until policy has been applied by Agent
+	require.Eventually(s.T(), func() bool {
+		getAgentReq := kibana.GetAgentRequest{ID: agent.ID}
+		updatedPolicyAgent, err := kibClient.GetAgent(getAgentReq)
+		require.NoError(s.T(), err)
+
+		return updatedPolicyAgent.PolicyRevision > prevAgentPolicyRevision
+	}, 30*time.Second, 1*time.Second)
```

Comment:
More time?

Reply:
Increased the timeout duration to 2 minutes in 60b06c3c9.